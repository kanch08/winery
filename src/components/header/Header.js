import React, {Component} from 'react';
import { connect } from 'react-redux';
import {NavLink} from "react-router-dom";
import cartimage from "../assets/cart.svg";
import NotificationBadge, {Effect} from "react-notification-badge";
import Slide from 'react-reveal/Slide';
import makeCarousel from 'react-reveal/makeCarousel';
import '../stylesheet/header.css';

class Header extends Component {
    render() {
        const { addedItems,offers }=this.props;
        const OfferContainer = (<div className="offer-section"> </div> );
        const CarouselUI = ({ children }) => <div className="offer-section">{children}</div>;


        const Carousel = makeCarousel(CarouselUI);


        return (

            <header>

                <Carousel defaultWait={1500} /*wait for 1000 milliseconds*/ >
                    {
                        offers.map((item,index)=>{
                            return(
                                <Slide bottom >
                                    <p key={item.id}>{item.text}</p>
                                </Slide>
                            )
                        })
                    }
                </Carousel>

                <div className="cart"
                     onMouseEnter={this.cartDetailOnOver}
                     onMouseLeave={this.cartDetailOnOver}
                >
                    <nav>

                    <NavLink to="/Cart">
                       <img
                        src={cartimage}
                        alt="cart"
                       />
                        <div className="badge container">
                            <NotificationBadge
                                count={addedItems.length || 0}
                                effect={Effect.ROTATE_X}
                            />
                        </div>
                    </NavLink>
                    </nav>

                </div>


            </header>
        );
    }
}

const mapStateToProps=(state)=>{
    return{
        offers:state.reducer.offers,
        addedItems:state.cartReducer.addedItems

    }
}

export default connect(mapStateToProps,null)(Header);