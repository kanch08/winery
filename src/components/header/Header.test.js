import Header from "./Header";
import { mount, shallow } from 'enzyme';
import React from "react";
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import toJson from "enzyme-to-json";

describe("testing Header component", () => {
    it("testing connected components", () => {
        const mockStore = configureStore();
        const initialState = {};
        const store = mockStore(initialState);
        const wrapper = shallow(<Provider store={store}><Header/></Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it("testing props", () => {
        const offers =[
            {
                "id":"201",
                "text":"Free shipping on prepaid-orders above INR 500"

            },
            {
                "id":"202",
                "text":"Great Offers from the Great Brands"
            },
            {
                "id":"203",
                "text":"Women Fashion Sale Coming Soon!!"
            }
        ];
        const mockStore = configureStore();
        const initialState = {};
        const store = mockStore(initialState);
        const wrapper = shallow(<Provider store={store}><Header offers={offers}/></Provider>);
        expect(wrapper.props().children.props.offers).not.toBe([]);
        // expect(wrapper.length).toEqual(1);
    })
})