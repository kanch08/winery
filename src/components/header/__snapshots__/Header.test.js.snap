// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header component testing connected components 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(Header) />
</ContextProvider>
`;
